src/Cli/Microsoft.DotNet.Cli.Utils/MSBuildForwardingAppWithoutLogging.cs

```diff
@@ -100,8 +99,6 @@ public void EnvironmentVariable(string name, string value)
 
             if (value == string.Empty || value == "\0")
             {
-                // Do not allow MSBuild NOIPROCNODE as null env vars are not properly transferred to build nodes
```

Comment:
We shall decide if same logic apply to msbuild server v2. See the comment at line 105.
Since we can't set env to null in reaction to Server build request message, we can't guarantee that build which runs in server mode will run exactly same as build which runs in out-of-proc as that build will have null env var thanks to `ProcessStartInfo.EnvironmentVariables`.
It is worth to notice that if success of build relay on null env var, such build can't be parallelized to MSBuild nodes as those have same restrictions and limitations as MSBuild server.